astropy/utils/iers/iers.py

```diff
@@ -802,3 +825,250 @@ def _substitute_iers_b(cls, table):
+        for f, allow_cache in trials:
+            if not allow_cache:
+                clear_download_cache(f)
+
+            try:
+                trial = cls.open(f, cache=True)
+            except Exception as exc:
+                err_list.append(exc)
+                continue
+
+            if self is None or trial.expires > best_expires:
+                self = trial
+                self.meta['data_url'] = str(f)
+                best_expires = trial.expires
+                if self.expires > good_enough:
+                    break
+
+        if self is None:
+            raise ValueError('none of the files could be read. The '
+                             'following errors were raised:\n' + str(err_list))
+
+        if self.expires < datetime.now():
+            warn('leap-second file is expired.', IERSStaleWarning)
+
+        return self
+
+    @property
+    def expires(self):
+        return self._expires
+
+    @classmethod
+    def _read_leap_seconds(cls, file, **kwargs):
+        """Read a leap-second table.
+
+        Finds data by removing comment lines, and identifies the
+        expiration date by matching with 'File expires'.
+        """
+        expires = None
+        # Find expiration date.
+        with get_readable_fileobj(file) as fh:
+            lines = fh.readlines()
+            for line in lines:
+                match = cls._re_expires.match(line)
+                if match:
+                    expires = datetime.strptime(match.groups()[0],
+                                                '%d %B %Y')
+                    break
+            else:
+                raise ValueError(f'did not find expiration date in {file}')
+
+        self = cls.read(lines, format='ascii.no_header', **kwargs)
+        self._expires = expires
+        return self
+
+    @classmethod
+    def from_iers_leap_seconds(cls, file=IERS_LEAP_SECOND_FILE):
+        return cls._read_leap_seconds(
+            file, names=['mjd', 'day', 'month', 'year', 'tai_utc'])
+
+    @classmethod
+    def from_leap_seconds_list(cls, file):
+        names = ['ntp_seconds', 'tai_utc', 'comment', 'day', 'month', 'year']
+        # Note: ntp_seconds does not fit in 32 bit, so causes problems on
+        # 32-bit systems without the np.int64 converter.
+        self = cls._read_leap_seconds(
+            file, names=names, include_names=names[:2],
+            converters={'col1': [convert_numpy(np.int64)]})
+        self['mjd'] = (self['ntp_seconds']/86400 + 15020).round()
+        # Note: cannot use Time in this routine.
+        dt = [datetime(1900, 1, 1) + timedelta(mjd-15020)
+              for mjd in self['mjd']]
+        self['day'] = [d.day for d in dt]
+        self['month'] = [d.month for d in dt]
+        self['year'] = [d.year for d in dt]
+        return self
+
+    @classmethod
+    def from_erfa(cls):
+        self = cls(erfa.leap_seconds.get())
+        self._expires = erfa.leap_seconds.expires
+        return self
+
```

Comment:
I was about to ask for this method but then realized it's pretty trivial to implement right here!  Basically the gist is: I think it's important to have an easy way to see what the "default baked in" version is, and this does that straightforwardly.

(It probably should get a test, though, if you accept this suggestion)

```suggestion
    @classmethod
    def from_builtin_erfa(cls):
        current = erfa.leap_seconds.get()
        try:
            erfa.leap_seconds.set(None)  # reset to defaults
            return cls.from_erfa()
        finally:
            erfa.leap_seconds.set(current)

```

Reply:
Good idea. Though this looses a possible expiration date, so I'll adjust a little.

Reply:
I've ended up doing it as an option for the `from_erfa` method.